thunder/core/transform_common.py

```diff
@@ -463,6 +463,119 @@ def is_functionalizable(bsym: BoundSymbol) -> bool:
     )
 
 
+def collect_required_copy_bsyms_for_args(
+    trace: Trace,
+    removed_copy_bsyms: list[BoundSymbol],
+) -> tuple[list[BoundSymbol], dict[VariableInterface, TensorProxy]]:
+    from thunder.core.utils import consumers as get_consumers
+    from thunder.core.utils import find_producer_symbols
+    from thunder.torch import _syms_returning_views
+
+    flat_args = tuple(
+        filter(
+            lambda p: isinstance(p, TensorProxy),
+            tree_flatten((trace.args, trace.kwargs))[0],
+        )
+    )
+    trace_args_set = ProxyDict()
+    for a in flat_args:
+        trace_args_set[a] = a
+
+    copy_dst_to_copy_bsym: dict[VariableInterface, BoundSymbol] = {}
+    required_copy_bsyms: list[BoundSymbol] = []
+    unused_copy_bsyms: list[BoundSymbol] = []
+    inplace_args: list[TensorProxy] = []
+    copy_destinations: list[TensorProxy] = []
+    for idx, bsym in enumerate(removed_copy_bsyms):
+        if (copy_to := bsym.flat_proxy_args[1]) in trace_args_set:
+            required_copy_bsyms.append(bsym)
+            copy_dst_to_copy_bsym[variableify(copy_to)] = bsym
+            inplace_args.append(copy_to)
+        else:
+            unused_copy_bsyms.append(bsym)
+
+        copy_destinations.append(copy_to)
+
+    consumers = get_consumers(trace)
+    view_args = []
+    for a in filter(lambda a: variableify(a) not in copy_dst_to_copy_bsym, flat_args):
+        first_consumer = consumers[a][0]
+        if first_consumer.sym in _syms_returning_views:
+            view_args.append(a)
+    stop_proxies = tuple(inplace_args + view_args)
+
+    new_copy_dst_to_copy_bsyms: dict[VariableInterface, BoundSymbol] = {}
+    additional_copy_bsyms: list[BoundSymbol] = []
+    swap_map: dict[VariableInterface, TensorProxy] = {}
+    for bsym in unused_copy_bsyms:
+        copy_return = bsym.flat_proxy_outs[0]
+
+        stop_proxies = flat_args if not inplace_args else inplace_args
+        s_bsyms = find_producer_symbols(trace, (copy_return,), stop_proxies)
+
+        first_bsym: BoundSymbol = s_bsyms[0]
+        if first_bsym.sym in _syms_returning_views:
+            if all(
+                (bsym.sym.tags and prims.OpTags.IN_PLACE in bsym.sym.tags) or bsym.sym in _syms_returning_views
+                for bsym in s_bsyms[1:]
+            ):
+                new_copy_dst: TensorProxy = list(filter(lambda p: isinstance(p, TensorProxy), first_bsym.flat_args))[0]
+                check(new_copy_dst in trace_args_set, lambda: f"{new_copy_dst=} not in {trace_args_set}")
+                copy_src: TensorProxy = bsym.flat_proxy_args[0]
```

Comment:
this does read maladroit to me because I do expect this check to does the equivalent or even the same as `elif` condition, except `new_copy_dst` check.
I however somehow fail to make it happen.

Also, the check should use `view_args` with its variables are `variableify`'ed.
```python
        if first_bsym.sym in _syms_returning_views:
            if all(
                (bsym.sym.tags and prims.OpTags.IN_PLACE in bsym.sym.tags) or bsym.sym in _syms_returning_views
                for bsym in s_bsyms[1:]
            ):
                new_copy_dst: TensorProxy = list(filter(lambda p: isinstance(p, TensorProxy), first_bsym.flat_args))[0]
                check(new_copy_dst in view_args, lambda: f"{new_copy_dst=} not in {view_args}")
                copy_src: TensorProxy = bsym.flat_proxy_args[0]
```

Reply:
Maybe we could add a comment so people looking won't be confused (I'm always confused, but more so when the branches of an if look too similar.)